### PR TITLE
Fix unit tests

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@vsadov, @dotnet/roslyn-compiler please review.

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs [hunk 1]
```diff
@@ -5221,27 +5221,24 @@ partial class C
 
             var compilation = CreateCompilationWithMscorlib(source, null, options: opt);
             compilation.VerifyDiagnostics(
-                // (25,25): error CS0759: No defining declaration found for implementing declaration of partial method 'C.Foo6<T>()'
-                //     static partial void Foo6<[A][A] T>() { }
-                Diagnostic(ErrorCode.ERR_PartialMethodMustHaveLatent, "Foo6").WithArguments("C.Foo6<T>()"),
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs]: Why is it Ok to drop this error?

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs [hunk 2]
```diff
@@ -5280,27 +5277,24 @@ partial class C
             var compilation = CreateCompilationWithMscorlib(source, null, options: opt);
 
             compilation.VerifyDiagnostics(
-                // (25,25): error CS0759: No defining declaration found for implementing declaration of partial method 'C.Foo6(int)'
-                //     static partial void Foo6([A][A] int y) { }
-                Diagnostic(ErrorCode.ERR_PartialMethodMustHaveLatent, "Foo6").WithArguments("C.Foo6(int)"),
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs]: Why is it Ok to drop this error?

File: src/Compilers/Core/Portable/Compilation/Compilation.cs [hunk 3]
```diff
@@ -1542,14 +1580,17 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                         filterOpt: null,
                         cancellationToken: cancellationToken);
 
-                    if (success)
+                    if (!moduleBeingBuilt.EmitOptions.EmitMetadataOnly)
                     {
-                        success = GenerateResourcesAndDocumentationComments(
+                        if (!GenerateResourcesAndDocumentationComments(
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Compilation/Compilation.cs]: Why is this the right thing to do when ```success``` is false?

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs (hunk 1 above)
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs]: This error should not be dropped. Restored in https://github.com/dotnet/roslyn/pull/10359.

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs (hunk 2 above)
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs]: Restored in https://github.com/dotnet/roslyn/pull/10359.

File: src/Compilers/Core/Portable/Compilation/Compilation.cs (hunk 3 above)
Comment (cston) [inline src/Compilers/Core/Portable/Compilation/Compilation.cs]: This matches the existing behavior in future: see `CSharpCompilation.CompileImpl`.

**Pull request merged**